sentry-android-okhttp/src/main/java/io/sentry/android/okhttp/SentryOkHttpInterceptor.kt

```diff
@@ -3,23 +3,37 @@ package io.sentry.android.okhttp
 import io.sentry.BaggageHeader
 import io.sentry.Breadcrumb
 import io.sentry.Hint
+import io.sentry.HttpStatusCodeRange
 import io.sentry.HubAdapter
 import io.sentry.IHub
 import io.sentry.ISpan
+import io.sentry.SentryEvent
 import io.sentry.SpanStatus
-import io.sentry.TracePropagationTargets
 import io.sentry.TypeCheckHint.OKHTTP_REQUEST
 import io.sentry.TypeCheckHint.OKHTTP_RESPONSE
+import io.sentry.exception.ExceptionMechanismException
+import io.sentry.exception.SentryHttpClientException
+import io.sentry.protocol.Mechanism
+import io.sentry.util.HttpUtils
+import io.sentry.util.PropagationTargetsUtils
+import okhttp3.Headers
 import okhttp3.Interceptor
 import okhttp3.Request
 import okhttp3.Response
 import java.io.IOException
 
 class SentryOkHttpInterceptor(
     private val hub: IHub = HubAdapter.getInstance(),
-    private val beforeSpan: BeforeSpanCallback? = null
+    private val beforeSpan: BeforeSpanCallback? = null,
+    // should this be under the options or here? also define the names
+    private val captureFailedRequests: Boolean = false,
+    private val failedRequestStatusCodes: List<HttpStatusCodeRange> = listOf(
+        HttpStatusCodeRange(HttpStatusCodeRange.DEFAULT_MIN, HttpStatusCodeRange.DEFAULT_MAX)
+    ),
+    private val failedRequestTargets: List<String> = listOf(".*")
```

Comment:
@romtsn is this going to break with https://docs.sentry.io/platforms/android/configuration/integrations/okhttp/#auto-installation-with-the-sentry-android-gradle-plugin ?
Do I need to add extra actors?

Reply:
As long as we keep the default no-arg constructor, it should be fine https://github.com/getsentry/sentry-android-gradle-plugin/blob/d59d745982b88649e2b89c5311a9f1a9c82dc15e/plugin-build/src/main/kotlin/io/sentry/android/gradle/instrumentation/okhttp/visitor/ResponseWithInterceptorChainMethodVisitor.kt#L48